fluss-lake/fluss-lake-paimon/src/test/java/com/alibaba/fluss/lake/paimon/tiering/PaimonTieringTest.java

```diff
@@ -169,74 +170,277 @@ void testTieringWriteTable(boolean isPrimaryKeyTable, boolean isPartitioned) thr
+        TablePath tablePath = TablePath.of("paimon", "test_three_partition");
+        createThreePartitionTable(tablePath);
+
+        Map<String, List<LogRecord>> recordsByPartition = new HashMap<>();
+        List<PaimonWriteResult> paimonWriteResults = new ArrayList<>();
+
+        // Test data for different three-level partitions
+        List<String> partitions =
+                Arrays.asList(
+                        "region=us-east/year=2024/month=01",
+                        "region=us-east/year=2024/month=02",
+                        "region=eu-central/year=2023/month=12");
+
+        int bucket = 0;
+
+        for (String partition : partitions) {
+            try (LakeWriter<PaimonWriteResult> lakeWriter =
+                    createLakeWriter(tablePath, bucket, partition)) {
+                List<LogRecord> logRecords =
+                        genLogTableRecordsForThreePartition(partition, bucket, 2);
+                recordsByPartition.put(partition, logRecords);
+
+                for (LogRecord logRecord : logRecords) {
+                    lakeWriter.write(logRecord);
+                }
+
+                PaimonWriteResult result = lakeWriter.complete();
+                paimonWriteResults.add(result);
+            }
+        }
+
+        // Commit all data
+        try (LakeCommitter<PaimonWriteResult, PaimonCommittable> lakeCommitter =
+                createLakeCommitter(tablePath)) {
+            PaimonCommittable committable = lakeCommitter.toCommitable(paimonWriteResults);
+            long snapshot = lakeCommitter.commit(committable);
+            assertThat(snapshot).isEqualTo(1);
+        }
+
+        // Verify data for each partition
+        for (String partition : partitions) {
+            List<LogRecord> expectRecords = recordsByPartition.get(partition);
+            CloseableIterator<InternalRow> actualRecords =
+                    getPaimonRowsThreePartition(tablePath, partition, bucket);
+            verifyLogTableRecordsThreePartition(actualRecords, expectRecords, bucket, partition);
+        }
+    }
+
     private void verifyLogTableRecords(
             CloseableIterator<InternalRow> actualRecords,
             List<LogRecord> expectRecords,
-            int expectBucket)
+            int expectBucket,
+            boolean isPartitioned,
+            @Nullable String partition)
             throws Exception {
         for (LogRecord expectRecord : expectRecords) {
             InternalRow actualRow = actualRecords.next();
             // check normal columns:
             assertThat(actualRow.getInt(0)).isEqualTo(expectRecord.getRow().getInt(0));
             assertThat(actualRow.getString(1).toString())
                     .isEqualTo(expectRecord.getRow().getString(1).toString());
-            assertThat(actualRow.getString(2).toString())
-                    .isEqualTo(expectRecord.getRow().getString(2).toString());
+
+            if (isPartitioned) {
```

Comment:
```suggestion
            if (partition != null) {
```
if `partition != null`, it means it's isPartitioned, so that we can remove the paramter `isPartitioned`
Actually, I believe we don't need to change this existing method.